File: backendApp/src/main/java/com/cinemas/entities/imageDescription.java
package com.cinemas.entities;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
@Table(name = "image_description")
public class imageDescription {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(unique = true, nullable = false)
    private Integer id;
    @Column
    private String url;
    @Column
    private String slug_name = null;
}

File: fllmbooking_app/lib/data/models/image_description.dart
class imageDescription {
  int? id;
  String? url;
  String? slug_name;

  imageDescription({this.id, this.url, this.slug_name});

  factory imageDescription.fromJson(Map<String, dynamic> json) {
    return imageDescription(
      id: json['id'],
      url: json['url'],
      slug_name: json['slug_name'],
    );
  }
}
